.github/workflows/java.yml

```diff
@@ -0,0 +1,106 @@
+name: Java
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/develop' }}
+
+on:
```

Comment:
So we basically release on every merge? How does versioning work here?

Reply:
versioning just uses git versioning. i'm not aware of any hard limits in sonatype. if we think it's too much, we can just do this on tag instead

Reply:
honestly IDK, if they won't ratelimit I don't care 🤷. The only issue is that its hard to align a vortex rust version which we should probably use more explicitly to communicate "quality" or just whether we made an effort to fix bugs.

Reply:
I've updated this so that we only run publish on tags. What I had before would not publish this as a SNAPSHOT, I think you need to add the `-SNAPSHOT` suffix for that to get picked up. I really just care about tag releases, so I think this is fine